src/main/protobuf/build/buildfarm/v1test/buildfarm.proto

```diff
@@ -1016,3 +1016,18 @@ service WorkerProfile {
 
   rpc GetWorkerList(WorkerListRequest) returns (WorkerListMessage) {}
 }
+
+message DrainWorkerPipelineRequest {
+  string instance_name = 1;
+
+  string worker_name = 2;
+}
+
+message DrainWorkerPipelineResults {
+  bool is_pipeline_empty = 1;
+}
+
+
+service ShutDownWorkerGracefully {
+  rpc DrainWorkerPipeline(DrainWorkerPipelineRequest) returns (DrainWorkerPipelineResults) {}
+}
```

Comment:
Is there a reason this is a separate service instead of part of the AdminService?

Reply:
yeah, that's what i want to do in the beginning.
The grpc call should be invoked from bf-admin or some other place(client side) and worker should be server side, which get informed and stop reregistering itself and finally give a response after everything is done. 

As far as I know, AdminService is actually the grpc server that hooked to bf scheduler right? so it cannot really inform a worker that it should stop reregistering itself?